draft-normington-jsonpath-latest.md

```diff
@@ -84,43 +108,266 @@ informative:
+| `/`   | `$`                | the root object/element                                                                                                                           |
+| `.`   | `@`                | the current object/element                                                                                                                        |
+| `/`   | `.` or `[]`        | child operator                                                                                                                                    |
+| `..`  | n/a                | parent operator                                                                                                                                   |
+| `//`  | `..`               | nested descendants. JSONPath borrows this syntax from E4X.                                                                               |
+| `*`   | `*`                | wildcard. All objects/elements regardless of their names.                                                                                         |
+| `@`   | n/a                | attribute access. JSON data items don't have attributes.                                                                                          |
+| `[]`  | `[]`               | subscript operator. XPath uses it to iterate over element collections and for predicates. In JavaScript and JSON it is the native array operator. |
+| `¦`   | `[,]`              | Union operator in XPath results in a combination of node sets. JSONPath allows alternate names or array indices as a set.                         |
+| n/a   | `[start:end:step]` | array slice operator borrowed from ES4.                                                                                                           |
+| `[]`  | `?()`              | applies a filter (script) expression.                                                                                                             |
+| n/a   | `()`               | script expression, using the underlying script engine.                                                                                            |
+| `()`  | n/a                | grouping in Xpath                                                                                                                                 |
+{: #tbl-overview title="Overview over JSONPath, comparing to XPath"}
+
+<!-- note that the weirdness about the vertical bar above is intentional -->
+
+XPath has a lot more to offer (location paths in unabbreviated syntax,
+operators and functions) than listed here.  Moreover there is a
+significant difference how the subscript operator works in Xpath and
+JSONPath:
+
+* Square brackets in XPath expressions always operate on the *node set* resulting from the previous path fragment. Indices always start at 1.
+* With JSONPath, square brackets operate on the *object* or *array* addressed by the previous path fragment. Indices always start at 0.
```

Comment:
JSON objects are specifically unordered, and many languages simply _can't_ reference them in the serialized sequence (Ruby comes to mind.)  We went over and over this recently in JSON Schema.

Reply:
The wording her certainly can be improved.  The existing text tries to emphasize that array indices start at 1 in XPath and at 0 in JSONPath, it certainly is not trying to invent numeric indices for maps (JSON objects).

Thanks for a good set of actionable proposals for improvement!  I'll submit an update on Wednesday.

Reply:
Long Wednesday.  Please check my proposed updates.

Reply:
I think specifying "array indices" is sufficient here. 👍